src/halmos/sevm.py

```diff
@@ -2394,17 +2450,26 @@ def run(self, ex0: Exec) -> Tuple[List[Exec], Steps, HalmosLogs]:
 
             except EvmException as err:
                 ex.halt(error=err)
-                out.append(ex)
+
+                if ex.callback is None:
+                    yield ex
+                else:
+                    yield from ex.callback(ex, stack, step_id)
+
                 continue
 
             except HalmosException as err:
                 if self.options["debug"]:
                     print(err)
+
                 ex.halt(data=None, error=err)
-                out.append(ex)
-                continue
 
-        return (out, steps, logs)
+                if ex.callback is None:
+                    yield ex
+                else:
+                    yield from ex.callback(ex, stack, step_id)
```

Comment:
can we extract these 4 lines in a helper function? They seem to be duplicated a couple times

a comment might help understanding the `yield from ex.callback(...)`, it takes some thinking to figure out 🧙‍♂️